dashboard/app/controllers/api/v1/sections_controller.rb

```
@@ -12,4 +12,50 @@ def index
   def show
     render json: @section.summarize
   end
+
+  # POST /api/v1/sections
+  # Create a new section
+  def create
+    # Validate login_type
+    # TODO: Push validation into model?
+    login_type = (['none', '', nil].include? params[:login_type]) ? 'email' : params[:login_type]
```

Do we use all these different ways to indicate 'none'? Can we limit the number of ways?

---

If we got rid of the `'none'` option, we'd be able to just check `.blank?` for '' and nil. This can also happen in the model, with a `before_validation` callback, or by overriding `level_type=` to convert those to `email`.

This could (should) be validated in the model (e.g. `validates_inclusion_of :level_type, in: ...`), and ideally the set of valid level types would be exposed in an array. 

Then we can rely on model validations instead of explicitly checking the fields in the controller, and `return head :bad_request unless section.valid?`.

---

In the pegasus API, `'none'` was the only valid way to indicate no login type.  I was hoping to normalize that a little without removing support for the old way of doing it.

Now that I think about it though, maybe it shouldn't be valid to create a section without a login type at all?  Especially now that we're keying single-table-inheritance on it. We can depend on the UI to provide defaults here, and reject requests that don't include it.

@joshlory thoughts?

---

Oh right, I forgot we're keying STI from level_type. So it is validated. 👍

---

It's... sort of validated that way.  An exception is raised when an invalid type is passed - it was uncaught, which was causing unit tests to fail even when I `expect_response :error` and would probably get reported out to Honeybadger (which I'd like to avoid).

---

Ah, in that case, it's worth adding an explicit validation so we can fail gracefully instead of logging to Honeybadger.

---

> Now that I think about it though, maybe it shouldn't be valid to create a section without a login type at all?

Agreed, this shouldn't be valid!